### ISSUE TITLE
PD refinement strategy

### ISSUE THREAD

**Issue by LluLl** [yzhouas/PD-Denoising-pytorch#3]
Hi,  I have read your paper. It' cool to use this pd denoiser. But I cannot understand the edffect of the 3,4 step, you just use this and say these steps can get the texture regions. Can you explain the effect of 3,4 step. Thanks

**Comment by yzhouas** [comment]
Thanks for your interests.
The 3-4 step is to remove the artifacts caused by Direct Pixel-shuffle Unsampling. It actually can be addressed by a simple convolutional layer. But in a real application, the stride number may not be 2, so it makes end-to-end training the convolution layer harder. Thus we conduct this refinement process to remove the artificial boundary. The ablation study is shown in Fig. 8.

**Comment by yzhouas** [comment]
> Hi, I have read your paper. It' cool to use this pd denoiser. But I cannot understand the edffect of the 3,4 step, you just use this and say these steps can get the texture regions. Can you explain the effect of 3,4 step. Thanks
Thanks for your interests.
The 3-4 step is to remove the artifacts caused by Direct Pixel-shuffle Unsampling. It actually can be addressed by a simple convolutional layer. But in a real application, the stride number may not be 2, so it makes end-to-end training the convolution layer harder. Thus we conduct this refinement process to remove the artificial boundary. The ablation study is shown in Fig. 8.

**Comment by yzhouas** [comment]
> Hi, I have read your paper. It' cool to use this pd denoiser. But I cannot understand the edffect of the 3,4 step, you just use this and say these steps can get the texture regions. Can you explain the effect of 3,4 step. Thanks

Thanks for your interests.
The 3-4 step is to remove the artifacts caused by Direct Pixel-shuffle Unsampling. It actually can be addressed by a simple convolutional layer. But in a real application, the stride number may not be 2, so it makes end-to-end training the convolution layer harder. Thus we conduct this refinement process to remove the artificial boundary. The ablation study is shown in Fig. 8.

**Comment by LluLl** [comment]
Thanks, I have another question about Figure4. For the step 5, the input is real noisy image in the figure. but in the code, the  input is image with pixelshuffle.

**Comment by yzhouas** [comment]
> Thanks, I have another question about Figure4. For the step 5, the input is real noisy image in the figure. but in the code, the input is image with pixelshuffle.

The results of these two settings are similar. There should be fewer artifacts if inputting real images directly. I will update that version later. Also, it seems there are some bugs when ps=3, and I will fix it later.

**Comment by LluLl** [comment]
Thanks. The denoiser in your paper is trained by the synthetic noisy. The synthetic noisy may be vary different with real noisy image. So for the step5, maybe it cannot fully denoise the noisy even if you lifte the noise estimation levels.

**Comment by yzhouas** [comment]
> Thanks. The denoiser in your paper is trained by the synthetic noisy. The synthetic noisy may be vary different with real noisy image. So for the step5, maybe it cannot fully denoise the noisy even if you lifte the noise estimation levels.

Yes, so that's why we try to first pixel-shuffle the images. In this case, the implementation version is more effective than the paper one. Actually, the arxiv one is not yet updated. We will further compare the difference between these two versions on more real image databases.

For CCD real noises on raw sensor data can be mixed of Gaussian-Poisson and Impulse Noise (for dead pixels) and after the in-camera pipeline to sRGB images, many complicated non-linear transforms happened. The noise level function (or the relationship between noise level with pixel original intensity) can be highly related to different 'gamma correction' like function (the CRF, gamma transform etc.) and interpolation(demosaicing) and clipping etc.. Even though most cameras will follow 'high-intensity pixel yields high-level noise', there are still many different cases. 
Modeling all the possibilities of noise distributions on all kinds of RGB images is almost impossible, so that's why we try to start from the spatial-correlation of noises. Using AWGN will be more suitable for general cases (like concave gamma correction transform will toggle the intensity-noise relationship), but there are not enough real noise datasets shot by different cameras to validate this point now. 

There is also one challenge to cope with those noisy images which have been resized. Pixel-shuffle may resolve this issue by increasing the stride number, but it also hurts the spatial-correlation of the original pixels. We are seeking a better way to model it.

But why we still need deep learning? Traditional methods do not actually care about noise distribution, but only similar patches or pixels. We still believe learning-based methods will yield better denoising performance with high-level semantic meaning modeling in the feature. 

So we still think PS may be interesting as a different viewpoint and a starting point of denoising, to cope with the 'most general' RGB denoising problems, but not just for some specific cameras.

**Comment by LluLl** [comment]
Thank you very much for your answer.
